test-app/tests/integration/components/combobox-test.gts

```diff
@@ -0,0 +1,854 @@
+   * input tag is blurred.  In that case, we want to clear the input
+   * and reset their selection to null.
+   */
+  test('it calls `@onChange` with null after a user clears their original input selection', async function (assert) {
+    assert.expect(4);
+
+    let handleChange = (value: unknown) => {
+      assert.strictEqual(value, null, 'Expected `value` to be null');
+      assert.step('handleChange');
+    };
+
+    // NOTE: We have a selected option
+    // NOTE: We add an input tag so we have something to blur to (by focusing another element)
+    await render(<template>
+      <ComboboxControl
+        @selected="blue"
+        @options={{testColors}}
+        @onChange={{handleChange}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+
+      {{! template-lint-disable require-input-label }}
+      <input placeholder="test" data-input />
+    </template>);
+
+    await click('[data-combobox]');
+
+    // Clear the input
+    await fillIn('[data-combobox]', '');
+
+    // Now blur the elment
+    await click('[data-input]');
+
+    assert.dom('[data-combobox]').hasValue('');
+    assert.verifySteps(['handleChange']);
+  });
+
+  test('it reverts to the selected option when a user enters garbage after previously having a valid selection (with `@selected` as a string)', async function (assert) {
+    assert.expect(3);
+
+    let handleChange = () => {
+      assert.step('do-not-expect-this-to-be-called!');
+    };
+
+    // NOTE: We have a selected option
+    // NOTE: We add an input tag so we have something to blur to (by focusing another element)
+    await render(<template>
+      <ComboboxControl
+        @selected="blue"
+        @options={{testColors}}
+        @onChange={{handleChange}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+
+      {{! template-lint-disable require-input-label }}
+      <input placeholder="test" data-input />
+    </template>);
+
+    await click('[data-combobox]');
+
+    // Enter garbage into the input
+    await fillIn('[data-combobox]', 'some-garbage');
+
+    // Now blur the elment
+    await click('[data-input]');
+
+    // Verify the input is reset to our `@selected` option
+    assert.dom('[data-combobox]').hasValue('blue');
+
+    // NOTE: We do not expect the `@onChange`  to be called in this
+    // case as we are only visually resetting to the previously
+    // selected value
+    assert.verifySteps([]);
+
+    // We want to verify the original options are re-displayed
+    // rather than the input being filtered to garbage
+    await click('[data-combobox]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+  });
+
+  test('it reverts to the selected option when a user enters garbage after previously having a valid selection (with `@selected` as an object)', async function (assert) {
```

Comment:
🎉  Love these exhaustive tests! Really gives me a lot of confidence in how to understand how this feature works.